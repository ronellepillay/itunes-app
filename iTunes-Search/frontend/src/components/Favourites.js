import React, { useState, useEffect } from 'react';
import { styled } from '@mui/material/styles';
import { List, ListItem, ListItemText, ListItemAvatar, Avatar, IconButton, Typography } from '@mui/material';
import DeleteIcon from '@mui/icons-material/Delete';

const StyledListItem = styled(ListItem)({
  maxWidth: 345,
  margin: '10px',
});

const StyledAvatar = styled(Avatar)({
  height: 140,
  width: 140,
});

const ListWrapper = styled('div')({
  display: 'flex',
  justifyContent: 'center',
});

const ListItemWrapper = styled('div')({
  display: 'flex',
  alignItems: 'center',
  '& .MuiListItemText-root': {
    marginLeft: '16px',
  },
});

export default function Favourites() {
    // State to store the list of favourites
    const [favourites, setFavourites] = useState([]);

    // Function to fetch the favourites list from the server
    const fetchFavourites = async () => {
        try {
            const response = await fetch('https://itunes-search.onrender.com/favourites');
            const data = await response.json();
            console.log(data); // add this line
            setFavourites(data);
        } catch (error) {
            console.error(error);
        }
    };

    // Load the favourites list from the server when the component mounts
    useEffect(() => {
        fetchFavourites();
    }, []);

    // Function to remove an item from the favourites list
    const handleRemoveFromFavourites = async (item) => {
        try {
            // Send a DELETE request to the server to remove the item from the favourites list
            const response = await fetch(`https://itunes-search.onrender.com/favourites/${item.trackId}`, {
                method: 'DELETE',
                headers: {
                    'Content-Type': 'application/json', 
                },
            });
            const data = await response.json();
            console.log(data.message);

            // Update the favourites list in the state
            const updatedFavourites = favourites.filter(
                (fav) => fav.trackId !== item.trackId
            );
            setFavourites(updatedFavourites);
        } catch (error) {
            console.error(error);
        }
    };

    return (
        <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', justifyContent: 'center' }}>
        <Typography variant="h4" component="h4" color="text.secondary" align='center'>
            Favourites
        </Typography>
        <ListWrapper>
            {favourites.length > 0 ? (
                <List>
                      {favourites.map((fav) => (
                            <StyledListItem>
                            <ListItemWrapper>
                                <ListItemAvatar>
                                    <StyledAvatar src={fav.artworkUrl100} alt={fav.trackName} />
                                </ListItemAvatar>
                                        <ListItemText primary={fav.trackName} secondary={fav.artistName} />
                                </ListItemWrapper>
                                <IconButton onClick={() => handleRemoveFromFavourites(fav)} aria-label="delete">
                                    <DeleteIcon />
                                </IconButton>
                            </StyledListItem>
                        ))}
                </List>
            ) : (
                <Typography variant="body2" color="text.secondary">
                    No Favourites yet.
                </Typography>
            )}
        </ListWrapper>
        </div>
    );
}

