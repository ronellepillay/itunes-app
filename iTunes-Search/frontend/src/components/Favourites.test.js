import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Favourites from './Favourites';

//Snapshot test
it('renders correctly', () => {
  const tree = renderer.create(<Favourites />).toJSON();
  expect(tree).toMatchSnapshot();
});

//Unit test 
describe('Favourites', () => {
  it('should render the Favourites component', () => {
    const { container } = render(<Favourites />);
    expect(container.firstChild).toMatchSnapshot();
  });
});