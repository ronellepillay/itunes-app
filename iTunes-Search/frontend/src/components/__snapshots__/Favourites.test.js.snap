// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourites should render the Favourites component 1`] = `
<div
  style="display: flex; flex-direction: column; align-items: center; justify-content: center;"
>
  <h4
    class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-9zqb3o-MuiTypography-root"
  >
    Favourites
  </h4>
  <div
    class="css-1l4w6pd"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
    >
      No Favourites yet.
    </p>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <h4
    className="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-9zqb3o-MuiTypography-root"
  >
    Favourites
  </h4>
  <div
    className="css-1l4w6pd"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
    >
      No Favourites yet.
    </p>
  </div>
</div>
`;
